### ISSUE TITLE
Platform Indicators Causing a loop of loading screens when enabled

### ISSUE THREAD

**Issue by SynatomicFox** [Strencher/BetterDiscordStuff#369]
As described in the title, as soon as you enable this plugin, discord starts looping with the loading screen, the only other plugin i have enabled is PinDM's by DevilBro and his plugin library.

**Comment by NicoMachine** [comment]
i found a version of ZeresPluginLibrary that works its version 2.0.19 i searched in the builds and the latest works https://github.com/rauenzi/BDPluginLibrary/actions/runs/6674121161

**Comment by NicoMachine** [comment]
i got it from the Artifacts link down the link and in Artifacts its in releases

**Comment by Technetium1** [comment]
[2.0.19](https://raw.githubusercontent.com/rauenzi/BDPluginLibrary/master/release/0PluginLibrary.plugin.js) from https://github.com/rauenzi/BDPluginLibrary/commit/99d49cd7a5041c8f7552e79df9cec88670cce36c brings the library back into working order, but it still doesn't solve the separate looping issue with platform indicators.

**Comment by Enrico9431** [comment]
Seems like the issue lies with this plugin as I also just found out that removing it stops the constant reloading

**Comment by ImAFrogOwO** [comment]
Nah Strencher has some weird undefined `_` reference that I can't seem to find where its defined.
For a fix replace `if (this.state && _.isEqual(this.state, this.getState())) return;` with `if (this.state /*&&_.isEqual(this.state, this.getState()) - This isnt even defined*/) return;`

**Comment by liigmabalz** [comment]
> Nah Strencher has some weird undefined `_` reference that I can't seem to find where its defined. For a (temp) fix replace
> 
> ```js
> if (this.state && _.isEqual(this.state, this.getState())) return;
> ```
> 
> with
> 
> ```js
> if (this.state /*&&_.isEqual(this.state, this.getState())*/) return;
> ```
> 
> The plugin disrupts the Gateway socket which causes weird reloading.

yo, thanks for the fix

**Comment by Knewest** [comment]
> Nah Strencher has some weird undefined `_` reference that I can't seem to find where its defined. For a (temp) fix replace
> 
> ```js
> if (this.state && _.isEqual(this.state, this.getState())) return;
> ```
> 
> with
> 
> ```js
> if (this.state /*&&_.isEqual(this.state, this.getState())*/) return;
> ```
> 
> The plugin disrupts the Gateway socket which causes weird reloading.

Can confirm this fixed the issue of loop reloading when DMs are opened.

**Comment by Strencher** [comment]
This is not a very idea fix, it will cause high resources usage. `_.isEqual` was a function provided by lodash which discord removed recently

**Comment by ImAFrogOwO** [comment]
I couldn't find what `_` was being defined as?

**Comment by ImAFrogOwO** [comment]
Alright, I believe to come up with a fix. My apologizes for that fix.

Using `isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});` believed to work? I am not sure if this is a fix you want.
```js
 isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});
                handleChange = () => {
                    if (this._destroyed) return false;

                    if (this.state && this.isEqual(this.state, this.getState())) return;

                    this.mount();
                }
```

**Comment by Knewest** [comment]
> Alright, I believe to come up with a fix. My apologizes for that fix.
> 
> Using `isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});` believed to work? I am not sure if this is a fix you want.
> 
> ```js
>  isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});
>                 handleChange = () => {
>                     if (this._destroyed) return false;
> 
>                     if (this.state && this.isEqual(this.state, this.getState())) return;
> 
>                     this.mount();
>                 }
> ```

I can confirm that this also fixes the loop loading error in DMs. It doesn't appear to cause any issues when using the plugin.

**Comment by Enrico9431** [comment]
> Alright, I believe to come up with a fix. My apologizes for that fix.
> 
> Using `isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});` believed to work? I am not sure if this is a fix you want.
> 
> ```js
>  isEqual = BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});
>                 handleChange = () => {
>                     if (this._destroyed) return false;
> 
>                     if (this.state && this.isEqual(this.state, this.getState())) return;
> 
>                     this.mount();
>                 }
> ```
> 
> Or maybe `BdApi.Webpack.getModule(x=>x.isEqual && x.is)` could work. I never used Lodash so my fix could be completely wrong.

I'm kinda confused now as to what exactly I gotta replace with what?

**Comment by Strencher** [comment]
I think @ImAFrogOwO solutions aren't very helpful. I don't have time to make a fix, sorry about that. Maybe @domi-btnr has time to make a quality fix.

**Comment by ImAFrogOwO** [comment]
Yeah so apparently I might not be helpful, but you might not be right either considering `BdApi.findModuleByProps("isEqual","add")` OR `BdApi.Webpack.getModule(x=>x.isEqual, {searchExports:true});` returns `lodash` which you said is removed by disccord.

Just trying to help out taking into consideration you don't have time.

**Comment by domi-btnr** [comment]
> I think @ImAFrogOwO solutions aren't very helpful. I don't have time to make a fix, sorry about that. Maybe @domi-btnr has time to make a quality fix.

`BdApi.findModuleByProps("isEqual","add")` seems to be the loadash `isEqual` function. Tested it, and it works without any Error nor Performance issues as far as I can tell. Fixed in [ac1c826](https://github.com/Strencher/BetterDiscordStuff/commit/ac1c82658329a589eebb217d2649b529427b16ed)
